std/experimental/allocator/package.d

```diff
@@ -1848,12 +2049,79 @@ CAllocatorImpl!(A, Yes.indirect) allocatorObject(A)(A* pa)
 
 /**
 
-Implementation of $(D IAllocator) using $(D Allocator). This adapts a
-statically-built allocator type to $(D IAllocator) that is directly usable by
+Returns a dynamically-typed $(D CSharedAllocator) built around a given statically-
+typed allocator $(D a) of type $(D A). Passing a pointer to the allocator
+creates a dynamic allocator around the allocator pointed to by the pointer,
+without attempting to copy or move it. Passing the allocator by value or
+reference behaves as follows.
+
+$(UL
+$(LI If $(D A) has no state, the resulting object is allocated in static
+shared storage.)
+$(LI If $(D A) has state and is copyable, the result will store a copy of it
+within. The result itself is allocated in its own statically-typed allocator.)
+$(LI If $(D A) has state and is not copyable, the result will move the
+passed-in argument into the result. The result itself is allocated in its own
+statically-typed allocator.)
+)
+
+*/
+shared(CSharedAllocatorImpl!A) sharedAllocatorObject(A)(auto ref A a)
+if (!isPointer!A)
+{
+    import std.conv : emplace;
+    static if (stateSize!A == 0)
+    {
+        enum s = stateSize!(CSharedAllocatorImpl!A).divideRoundUp(ulong.sizeof);
```

Comment:
`ulong` is used here as the data type with the largest alignment. What's a better choice?